src/hades/agent/signature.py

```diff
@@ -0,0 +1,132 @@
+            raise ValueError("Content encoding of inner serializer {!r} must "
+                             "match ({!r} != {!r})"
+                             .format(self._serializer, content_encoding,
+                                     self._content_encoding))
+        body = self._ensure_bytes(body)
+        if len(body) > 0 and body[0] in self.whitespace:
+            raise ValueError("Inner data may not begin with the following "
+                             "characters {!r}"
+                             .format(str(self.whitespace)))
+        message = self._signing_key.sign(body)
+        signature = self.key_codec.encode(message.signature).decode('ascii')
+        header = {
+            'signature':        signature,
+            'signer':           self._signer,
+            'content_type':     content_type,
+            'content_encoding': content_encoding,
+        }
+        buffer = io.BytesIO()
+        wrapper = io.TextIOWrapper(buffer, self._content_encoding,
+                                   write_through=True)
+        with wrapper:
+            json.dump(header, wrapper)
+            buffer.write(b"\n")
+            buffer.write(message.message)
+            return buffer.getvalue()
+
+    def parse_header(self, data):
+        return self.json_decoder.raw_decode(data.decode(self._content_encoding))
+
+    def deserialize(self, data):
+        data = self._ensure_bytes(data)
+        header, end = self.parse_header(data)
+        # Skip whitespace
+        length = len(data)
+        while end < length and data[end] in self.whitespace:
+            end += 1
+        header, body = header, data[end:]
+
+        signer, signature, content_type, content_encoding = (
+            header['signer'], header['signature'],
+            header['content_type'], header['content_encoding']
+        )
+        signature = self.key_codec.decode(signature)
+        if content_encoding != self._content_encoding:
+            raise ValueError("Invalid inner content encoding ({!r} != {!r})"
+                             .format(content_encoding, self._content_encoding))
+
+        try:
+            verify_key = self._verify_keys[signer]
+        except KeyError:
+            raise ValueError("Unknown signer {!r}".format(signer)) from None
+        verify_key.verify(body, signature)
+        return loads(bytes_to_str(body), content_type, content_encoding,
+                     force=True)
+
+
+def register(signing_key: nacl.signing.SigningKey,
+             verify_keys: Iterable[nacl.signing.VerifyKey],
+             name: str = 'ed25519', serializer='json',
+             content_type: str = 'application/x-data-ed25519',
+             content_encoding: str = 'utf-8'):
+    """Register serializer with :mod:`kombu`"""
+    s = ED25519Serializer(signing_key, verify_keys, serializer,
+                          content_encoding)
+    kombu_register(name, s.serialize, s.deserialize, content_type,
+                   content_encoding)
```

Comment:
How precisely would this serializer be enabled? I don't see any config changes in this PR.